### PR TITLE
Copy action constraints when setting up a PageActionDescriptor

### LINKED ISSUE #8207: ActionConstraints added to PageRouteModel.SelectorModel do not get copied to the action descriptor

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #8207

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs [hunk 1]
```diff
@@ -75,6 +75,7 @@ private void AddActionDescriptors(IList<ActionDescriptor> actions, PageRouteMode
             {
                 var descriptor = new PageActionDescriptor
                 {
+                    ActionConstraints = selector.ActionConstraints,
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs]: Make sense. However, this initialization also doesn't set at least the `EndpointMetadata`, `BoundProperties` or `Parameters` properties. Should it?

File: test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Infrastructure/PageActionDescriptorProviderTest.cs [hunk 2]
```diff
@@ -167,6 +168,41 @@ public void GetDescriptors_CopiesRouteValuesFromModel()
...
+            var context = new ActionDescriptorProviderContext();
+
+            // Act
+            provider.OnProvidersExecuting(context);
+
+            // Assert
+            var result = Assert.Single(context.Results);
+            var descriptor = Assert.IsType<PageActionDescriptor>(result);
+            Assert.Equal(model.RelativePath, descriptor.RelativePath);
+            Assert.Collection(
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Infrastructure/PageActionDescriptorProviderTest.cs]: Minor but it's weird to use `Assert.Collection(...)` for a 1-entry collection. Use `Assert.Same(expected, Assert.Single(descriptor.ActionConstraints))` or (if you prefer) split the assertions.

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs (hunk 1 above)
Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs]: It was missing `EndpointMetadata`. We don't calculate `BoundProperties` or `Parameters` until a request comes in and we've compiled the page.

**Comment by pranavkm** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs [hunk 3]
```diff
@@ -75,6 +75,8 @@ private void AddActionDescriptors(IList<ActionDescriptor> actions, PageRouteMode
             {
                 var descriptor = new PageActionDescriptor
                 {
+                    ActionConstraints = selector.ActionConstraints.ToList(),
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs]: Why `ToList()`? In particular, why use `ToList()` in two of the the three additional copies?

**Pull request merged**